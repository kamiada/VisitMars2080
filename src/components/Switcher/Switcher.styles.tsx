import styled from 'styled-components';

export const Container = styled.div`
  float: left;
`;
export const SwitchButton = styled.button`
  color: #fff;
  font-size: 1em;
  margin: 1em;
  border: transparent;
  border-radius: 3px;
  background: transparent;
  font-family: 'Libre Baskerville', serif;
  font-size: 30px;
`;