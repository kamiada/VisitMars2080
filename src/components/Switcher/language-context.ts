import * as React from "react";

export const languages = {
    "ENG": [
        { "exampleLine":"Hello World"}
     ],
     "FR": [
         {
             "exampleLine":"Bonjour le monde"
         }
     ],
     "PL": [
         {
             "exampleLine":"Witaj Åšwiecie"
         }
     ]
  };
  
  export const LanguageContext = React.createContext(languages.ENG);