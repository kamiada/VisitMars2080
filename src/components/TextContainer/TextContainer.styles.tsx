import styled from 'styled-components';
import * as palette from '../../variables';
export const Container = styled.div`
  width: 300px;
  height: 400px;
  border: 1px white solid;
  background-color: ${palette.creamWhite};
  color: ${palette.notexactlyblack};
  text-align: justify;
  padding: 20px;
  margin: 10px;
  border-radius: 10px;
  top: 20%;
  position: absolute;
  line-height: 1.5;
  opacity: 0.9;
`;