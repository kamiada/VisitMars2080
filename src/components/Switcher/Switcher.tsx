import * as React from 'react';
import { Container, SwitchButton } from './Switcher.styles';
import { LanguageContext } from './language-context';


export const Switcher: React.FC<{}> = props => {
    const langContext = React.useContext(LanguageContext);
    return (
        <Container>
            <SwitchButton>
                {props}
                {langContext}
            </SwitchButton>
        </Container>
    )
}
