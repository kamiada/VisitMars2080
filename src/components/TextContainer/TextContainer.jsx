import React, {useContext} from 'react';
import { Container } from './TextContainer.styles';
import { LanguageContext, dictionaryList } from '../Switcher/language-context';


export const TextContainer = ({id}) => {
    const { language } = useContext(LanguageContext);
    return (
        <Container >
            <p>{dictionaryList[language][id]}</p>
        </Container>
    )
}