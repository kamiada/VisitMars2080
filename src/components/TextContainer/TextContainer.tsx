import * as React from 'react';
import { Container } from './TextContainer.styles';
import { FC } from 'react';

interface Props {
    text: String
}

export const TextContainer: FC<Props> = ({text}) => {
    return (
        <Container >
            {text}
        </Container>
    )
}