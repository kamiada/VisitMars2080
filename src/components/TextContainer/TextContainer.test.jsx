/* eslint-disable testing-library/render-result-naming-convention */
import { render } from "@testing-library/react";
import * as React from "react";
import { TextContainer } from "./TextContainer";
import { LanguageContext } from "../Switcher/language-context";

const language = {
  language: "ENG",
  toggleLanguage: jest.fn(),
};

describe("TextContainer", () => {
  test("Matches snapshot - default", () => {
    const renderTextContainer = render(
      <LanguageContext.Provider value={language}>
        <TextContainer id="welcomeText" />
      </LanguageContext.Provider>
    );
    expect(renderTextContainer).toMatchSnapshot();
  });
  test("Matches snapshot - French", () => {
    const language = {
      language: "FR",
      toggleLanguage: jest.fn(),
    };
    const renderTextContainer = render(
      <LanguageContext.Provider value={language}>
        <TextContainer id="welcomeText" />
      </LanguageContext.Provider>
    );
    expect(renderTextContainer).toMatchSnapshot();
  });
  test("Matches snapshot - Polish", () => {
    const language = {
      language: "PL",
      toggleLanguage: jest.fn(),
    };
    const renderTextContainer = render(
      <LanguageContext.Provider value={language}>
        <TextContainer id="welcomeText" />
      </LanguageContext.Provider>
    );
    expect(renderTextContainer).toMatchSnapshot();
  });
});
