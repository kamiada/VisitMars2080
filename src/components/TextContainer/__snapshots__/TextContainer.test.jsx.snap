// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContainer Matches snapshot - French 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ zVsQE"
      >
        <p>
          Bienvenue sur Mars ! Vous pouvez enfin réserver les vacances de vos rêves avec nos forfaits ultra luxueux, conçus rien que pour vous ! Laissez-vous oublier les problèmes du réchauffement climatique, du capitalisme, du choc des classes et tous les autres problèmes, en vous retirant tout simplement de la Terre. Cliquez sur la page suivante pour choisir une date, puis choisissez la bonne façon de vous gâter pendant que vous êtes sur la planète rouge et envoyez-nous enfin tout l'argent, dont vous n'avez absolument pas besoin. Vous méritez de vous faire plaisir !
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ zVsQE"
    >
      <p>
        Bienvenue sur Mars ! Vous pouvez enfin réserver les vacances de vos rêves avec nos forfaits ultra luxueux, conçus rien que pour vous ! Laissez-vous oublier les problèmes du réchauffement climatique, du capitalisme, du choc des classes et tous les autres problèmes, en vous retirant tout simplement de la Terre. Cliquez sur la page suivante pour choisir une date, puis choisissez la bonne façon de vous gâter pendant que vous êtes sur la planète rouge et envoyez-nous enfin tout l'argent, dont vous n'avez absolument pas besoin. Vous méritez de vous faire plaisir !
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TextContainer Matches snapshot - Polish 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ zVsQE"
      >
        <p>
          Witamy na Marsie! Już teraz możesz wybrać dla siebie wakacje swojego życia, zaprojektowanie tylko dla Ciebie! Pozwól sobie zapomnieć o globalnym ociepleniu, kapitalizmie, walce klas i wszystkich tych ziemskich problemach, poprzez jakże proste opuszczenie Ziemi. Kliknij na następną stronę, wybierz odpowiednią dla siebie datę, a następnie wybierz sposób w jaki planujesz się rozpieścić na Czerwonej Planecie. Następnie wyślij nam wszystkie te pieniądze, których z pewnością nie potrzebujesz. Zasługujesz na to by się rozpieścić!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ zVsQE"
    >
      <p>
        Witamy na Marsie! Już teraz możesz wybrać dla siebie wakacje swojego życia, zaprojektowanie tylko dla Ciebie! Pozwól sobie zapomnieć o globalnym ociepleniu, kapitalizmie, walce klas i wszystkich tych ziemskich problemach, poprzez jakże proste opuszczenie Ziemi. Kliknij na następną stronę, wybierz odpowiednią dla siebie datę, a następnie wybierz sposób w jaki planujesz się rozpieścić na Czerwonej Planecie. Następnie wyślij nam wszystkie te pieniądze, których z pewnością nie potrzebujesz. Zasługujesz na to by się rozpieścić!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TextContainer Matches snapshot - default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ zVsQE"
      >
        <p>
          Welcome to Mars! Now you can finally book the holidays of your dream with our ultra luxurious packages, designed just for you! Let yourself forgot about the problems of global warming, capitalism, clash of classes and all the other problems, by simply removing yourself from the Earth. Click onto next page, to pick a date, then chose right form of spoiling yourself while on the Red Planet and finally send us all the money, which you totally don't need. You deserve to treat yourself!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ zVsQE"
    >
      <p>
        Welcome to Mars! Now you can finally book the holidays of your dream with our ultra luxurious packages, designed just for you! Let yourself forgot about the problems of global warming, capitalism, clash of classes and all the other problems, by simply removing yourself from the Earth. Click onto next page, to pick a date, then chose right form of spoiling yourself while on the Red Planet and finally send us all the money, which you totally don't need. You deserve to treat yourself!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
